smslibrary/src/main/java/com/eis/communication/Payload.java

```diff
@@ -0,0 +1,61 @@
+package com.eis.communication;
+
+import com.eis.communication.exception.InvalidPayloadException;
+
+/**
+ * Payload abstract class
+ * @author Mattia Fanan
+ *
+ * @param T Payload data data-type
+ */
+public abstract class Payload<T> {
+    protected T data;
+
+    /**
+     * Create a payload
+     * @param data Data of the Payload
+     * @throws InvalidPayloadException when an invalid data is passed
+     */
+    public Payload(T data)throws InvalidPayloadException {
+        if(data==null || !isValidData(data))
+            throw new InvalidPayloadException();
+
+        this.data = data;
+    }
+
+    /**
+     * Get data of the Payload
+     * @return data of the payload
+     */
+    public T getData() {
+        return data;
+    }
+
+    /**
+     * Set data of the Payload
+     * @param data Data to assign to the payload
+     * @return true if data is valid, false if data is not valid
+     */
+    public boolean setData(T data) {
+        if(data==null || !isValidData(data))
+            return false;
+
+        this.data = data;
+        return true;
+    }
+
+    /**
+     * Check if Payload is valid
+     * @return true if Payload is valid, false if Payload is not valid
+     */
+    public boolean isValid(){
```

Comment:
Can I have an invalid instance of Payload? If no, what is the purpose of this method? Will it ever return false?

Reply:
👍
I have removed all isValid methods
(In my repo)